featurebyte/models/feature.py

```diff
@@ -161,11 +198,40 @@ class FrozenFeatureModel(FeatureByteCatalogBaseDocumentModel):
     graph: QueryGraph = Field(allow_mutation=False)
     node_name: str
     tabular_source: TabularSource = Field(allow_mutation=False)
+    readiness: FeatureReadiness = Field(allow_mutation=False, default=FeatureReadiness.DRAFT)
+    version: VersionIdentifier = Field(
+        allow_mutation=False,
+        default=None,
+        description="Returns the version identifier of a Feature object.",
+    )
+    online_enabled: bool = Field(allow_mutation=False, default=False)
+    definition: Optional[str] = Field(
+        allow_mutation=False, default=None, description="Feature Definition"
+    )
+
+    # list of IDs attached to this feature
     entity_ids: List[PydanticObjectId] = Field(allow_mutation=False, default_factory=list)
     table_ids: List[PydanticObjectId] = Field(allow_mutation=False, default_factory=list)
     primary_table_ids: List[PydanticObjectId] = Field(allow_mutation=False, default_factory=list)
     feature_namespace_id: PydanticObjectId = Field(allow_mutation=False, default_factory=ObjectId)
     feature_list_ids: List[PydanticObjectId] = Field(allow_mutation=False, default_factory=list)
+    deployed_feature_list_ids: List[PydanticObjectId] = Field(
+        allow_mutation=False, default_factory=list
+    )
+
+    # pydantic validators
+    _version_validator = validator("version", pre=True, allow_reuse=True)(version_validator)
+
+    @root_validator
+    @classmethod
+    def _add_derived_attributes(cls, values: dict[str, Any]) -> dict[str, Any]:
+        # extract table ids & entity ids from the graph
+        graph = values["graph"]
+        node_name = values["node_name"]
+        values["primary_table_ids"] = graph.get_primary_table_ids(node_name=node_name)
+        values["table_ids"] = graph.get_table_ids(node_name=node_name)
+        values["entity_ids"] = graph.get_entity_ids(node_name=node_name)
+        return values
```

Comment:
The logic here is cleaner as we don't need to consider the case where this constructor is triggered by feature api object.